### PR TITLE
Feature/knit menu poc

### PR DESCRIPTION
**Pull request by FieryCod**
Adds knitMenu and knitFooter components for desktop.

### CODE REVIEW

File: components/Logo.vue [hunk 1]
```diff
@@ -1,79 +0,0 @@
...
-.Triangle--three {
-  top: 60px;
-  left: 35px;
-  animation: goright 0.5s linear forwards 3.5s;
-  border-left: 70px solid transparent;
-  border-right: 70px solid transparent;
-  border-bottom: 120px solid #35495E;
-}
-
-.Triangle--four {
```
Comment (k911) [inline components/Logo.vue]: In BEM I think there are not capital letters.

File: components/desktop/KnitMenu.vue [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    }, 500)
+  },
+  mounted () {
+    this.onResize()
+    this.toggleWideMenu()
+    window.addEventListener('scroll', this.toggleWideMenu)
+    window.addEventListener('resize', this.onResize)
+  },
+  destroyed () {
+    window.removeEventListener('scroll', this.toggleWideMenu)
```
Comment (k911) [inline components/desktop/KnitMenu.vue]: You forgot to remove one listener.

File: middleware/isMobile.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+export default function (ctx) {
+  const userAgent = ctx.req ? ctx.req.headers['user-agent'] : navigator.userAgent
+  ctx.isMobile = /Mobile|iP(hone|od|ad)|Android|BlackBerry|IEMobile|Kindle|NetFront|Silk-Accelerated|(hpw|web)OS|Fennec|Minimo|Opera M(obi|ini)|Blazer|Dolfin|Dolphin|Skyfire|Zune/i.test(userAgent)
```
Comment (k911) [inline middleware/isMobile.js]: Can you change:

```/Mobile|iP(hone|od|ad)|Android|BlackBerry|IEMobile|Kindle|NetFront|Silk-Accelerated|(hpw|web)OS|Fennec|Minimo|Opera M(obi|ini)|Blazer|Dolfin|Dolphin|Skyfire|Zune/i```

To: 

```/Mobile|iP(?:hone|od|ad)|Android|BlackBerry|IEMobile|Kindle|NetFront|Silk-Accelerated|(?:hpw|web)OS|Fennec|Minimo|Opera M(?:obi|ini)|Blazer|Dolfin|Dolphin|Skyfire|Zune/i```

I've added non-capturing groups which makes this regex faster.

File: components/desktop/KnitMenu.vue [hunk 4]
```diff
@@ -0,0 +1,223 @@
+<template>
+  <nav class="knitMenu" :class="{knitHighMenu: isHighMenu, knitShortMenu: !isHighMenu}">
+    <div class="knitLogotypeWrapper">
+      <img class="knitLogotype" src="knit-logotype.svg" alt="logo KNIT">
```
Comment (zrebcu411) [inline components/desktop/KnitMenu.vue]: A website logo in every website should be wrapped to `h1` and link to the main page. An `alt` attribute of this image should be KNIT or KNIT homepage, without 'logo' part.

File: components/desktop/KnitMenu.vue [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+    </div>
+    <div class="knitMenuBlockWrapper">
+      <div class="organization">
+        <h3>Koło Naukowe IT</h3>
```
Comment (zrebcu411) [inline components/desktop/KnitMenu.vue]: You are using h3 heading without h1 and h2 on the page. It breaks w3c rules.

**Comment by zrebcu411** [comment]
You don't use any styling convention in all your styles. We should choose one (OOCSS, BEM or SMACSS). I suggest rewriting all classes to BEM and  there are some things to change:  
- we shouldn't style by tags like `span` or `h3`, every element should have own class. Why? What about if you decide to change h3 to h2? You will have to change scss file too.
- use `rem` unit everywhere except things like border-radius, box-shadow which have to have a fixed value. Why? What about if a user set the bigger font size in browser options? All text will scale up, but sizes of elements not. It will destroy all website layout.

File: components/desktop/KnitMenu.vue [hunk 6]
```diff
@@ -0,0 +1,218 @@
+<template>
+  <header class="knit-menu" :class="{'knit-menu--high': isHighMenu, 'knit-menu--low': !isHighMenu}">
```
Comment (zrebcu411) [inline components/desktop/KnitMenu.vue]: Maybe you should change 'knit-menu' to 'knit-header'?

File: components/desktop/KnitMenu.vue [hunk 7]
```diff
@@ -0,0 +1,218 @@
...
+  &__nav-link {
+    margin: 0 10px;
+    white-space: nowrap;
+    text-transform: uppercase;
+    text-decoration: none;
+    color: $knit-menu-links-color;
+    &:hover {
+      border-bottom: 1px solid $knit-menu-links-hover-color;
+    }
+    &:active {
```
Comment (zrebcu411) [inline components/desktop/KnitMenu.vue]: You have to add focus state. See focus behavior using the Tab key.

**Pull request merged**